app/src/main/java/io/gnosis/safe/ui/base/BaseViewBindingFragment.kt

```diff
@@ -0,0 +1,37 @@
+package io.gnosis.safe.ui.base
+
+import android.content.Context
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.viewbinding.ViewBinding
+
+abstract class BaseViewBindingFragment<T> : BaseFragment()
+        where T : ViewBinding {
+
+    private var _binding: T? = null
+
+    protected val binding get() = _binding!!
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        inject(buildViewComponent(context))
```

Comment:
Injection is not totally decoupled like you mentioned. It seems like you are injecting the fragment still here.

Reply:
why should it be totally decoupled? I have not mentioned that